eosclubhouse/clubhouse.py

```diff
@@ -3088,18 +3091,42 @@ def __init__(self):
         self._init_style()
         InAppNotify.init_message()
 
+        installed = Desktop.is_hack_extension_installed(isEnabled=True)
+        USE_INAPP_NOTIFICATIONS = not installed
+        self._installing_extension = not installed
+
         # This will set the hack mode for old EOS < 3.9
         Desktop.set_legacy_hack_mode(True)
 
-        if not Desktop.is_hack_extension_installed(isEnabled=True):
-            Desktop.install_hack_extension(callback=self._on_extension_installed)
+    def _install_extension(self, action=None, arg_variant=None):
+        self._installing_extension = True
+        Desktop.install_hack_extension(callback=self._on_extension_installed)
+
+    def _show_extension_button(self, show=True):
+        if not self._window:
+            return
+
+        layer = self._window.clubhouse.get_layer(ClubhouseView.MAIN_LAYER_NAME)
+        if show:
+            layer.extension_button.show()
+        else:
+            layer.extension_button.hide()
```

Comment:
It would be better to use events for this. You can do it with a gobject property:

    extension_installed = GObject.Property(type=bool)

Or maybe:

    @GObject.Property(type=bool)
    def extension_installed(self):
        return Desktop.is_hack_extension_installed(isEnabled=True)

And then bind the button `visible` property to this property.

Reply:
I've added a new commit to use properties for this and for the `USE_INAPP_NOTIFICATIONS`, so we can use the binding and notify signals.